test/libsolidity/util/TestFileParser.h

```diff
@@ -0,0 +1,195 @@
+ * arguments, an optional ether value and an expected execution result.
+ */
+struct FunctionCall
+{
+	std::string signature;
+	FunctionCallArgs arguments;
+	FunctionCallExpectations expectations;
+	u256 value;
+};
+
+/**
+ * Class that is able to parse an additional and well-formed comment section in a Solidity
+ * source file used by the file-based unit test environment. For now, it parses function
+ * calls and their expected result after the call was made.
+ *
+ * - Function calls defined in blocks:
+ * // f(uint256, uint256): 1, 1 # Signature and comma-separated list of arguments
+ * // -> 1, 1                   # Expected result value
+ * // g(), 2 ether              # (Optional) Ether to be send with the call
+ * // -> 2, 3
+ * // h(uint256), 1 ether: 42
+ * // REVERT
+ * ...
+ */
+class TestFileParser
+{
+public:
+
+	/// Tries to convert the formatted \param _string to it's byte representation and
+	/// preserves the chosen byte formats. Supported types:
+	/// - unsigned and signed decimal number literals
+	/// Throws an InvalidArgumentEncoding exception if data is encoded incorrectly or
+	/// if data type is not supported.
+	static std::pair<bytes, ByteFormats> formattedStringToBytes(std::string _string);
+
+	/// Formats \param _bytes given the byte formats \param _formats. Supported formats:
+	/// - unsigned and signed decimal number literals
+	static std::string bytesToFormattedString(bytes const& _bytes, ByteFormats const& _formats);
+
+	/// Constructor that takes an input stream \param _stream to operate on
+	/// and creates the internal scanner.
+	TestFileParser(std::istream& _stream): m_scanner(_stream) {}
+
+	/// Parses function calls blockwise and returns a list of function calls found.
+	/// Throws an exception if a function call cannot be parsed because of its
+	/// incorrect structure, an invalid or unsupported encoding
+	/// of its arguments or expected results.
+	std::vector<FunctionCall> parseFunctionCalls();
+
+private:
+	/**
+	 * Simple scanner that is used internally to abstract away character traversal.
+	 */
+	class Scanner
+	{
+	public:
+		/// Constructor that takes an input stream \param _stream to operate on.
+		Scanner(std::istream& _stream): m_stream(_stream) {}
+
+		/// Returns the current character.
+		char current() const { return *m_char; }
+		/// Returns true if the end of a line is reached, false otherwise.
+		bool eol() const { return m_char == m_line.end(); }
+		/// Returns an iterator of the current position in the input stream.
+		std::string::iterator& position() { return m_char; }
+		/// Returns an iterator to the end position of the input stream.
+		std::string::iterator endPosition() { return m_line.end(); }
+		/// Advances current position in the input stream.
+		void advance() { ++m_char; }
+		/// Advances stream by one line. Returns true if stream contains a new line,
+		/// false if no new line could be read.
+		bool advanceLine()
+		{
+			std::istream& line = std::getline(m_stream, m_line);
```

Comment:
Is `line` different from `m_stream`? If not, please use `m_stream` to avoid such confusion :)